### DISCUSSION TITLE
How to implement a TextBox with the text distributed uniformly (like a Combs of characters in a PDF form) ?

### DISCUSSION THREAD

**Discussion by Whiletru3** [AvaloniaUI/Avalonia#15164] [Q&A]
Hi,

In PDFs, a form field has the notion of Combs of characters (check this link : https://evermap.com/Tutorial_AMM_Using_Combs_Of_Characters_In_PDF_Forms.asp )

I can't find an elegant way to implement this in AvaloniaUI other than simulate a textbox for each characters placed uniformly on a canvas. Wich is pain in the a$$ to use, set/set value, maintain, ...

Is there a better way ? If not this could be a good improvement 

Thanks :)

**Comment by timunie** [comment]
1. use a Font with monospaced chars
2. maybe set letter spacing in addition
```xml
<TextBox FontFamily="Courier New" LetterSpacing="5" Text="123" />
```

**Reply by wsficke** [reply]
One generic solution would be to implement your own font.  Make SVG for each letter, use font-making software to put it together, and import like any other font.

https://www.wikihow.com/Create-a-Font

**Reply by Whiletru3** [reply]
good, but unfortunately, I'm still in 0.10.x, no backport of LetterSpacing in it :/
I plan to move to 11 soon, so I'll wait :)
